internal/provider/kubernetes/controller.go

```diff
@@ -383,8 +383,9 @@ func (r *gatewayAPIReconciler) processBackendRefs(ctx context.Context, gwcResour
 					"name", string(backendRef.Name))
 			} else {
 				resourceMappings.allAssociatedNamespaces.Insert(serviceImport.Namespace)
-				if !resourceMappings.allAssociatedServiceImports.Has(utils.NamespacedName(serviceImport).String()) {
-					resourceMappings.allAssociatedServiceImports.Insert(utils.NamespacedName(serviceImport).String())
+				key := utils.NamespacedName(serviceImport).String()
```

Comment:
I prefer this style, but I found and ref other places, so the last version is more utils to get the `NamespacedName`